mount/mount_linux.go

```diff
@@ -17,16 +17,37 @@
 package mount
 
 import (
+	"fmt"
+	"os"
+	"path"
 	"strings"
 	"time"
 
+	"github.com/containerd/containerd/sys"
 	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
+var pagesize = 4096
+
+func init() {
+	pagesize = os.Getpagesize()
+}
+
 // Mount to the provided target path
 func (m *Mount) Mount(target string) error {
-	flags, data := parseMountOptions(m.Options)
+	var (
+		chdir   string
+		options = m.Options
+	)
+
+	// avoid hitting one page limit of mount argument buffer
+	//
+	// NOTE: 512 is a buffer during pagesize check.
+	if m.Type == "overlay" && optionsSize(options) >= pagesize-512 {
+		chdir, options = compactLowerdirOption(options)
```

Comment:
Yet better, move this check to after parseMountOptions() as it's simpler and more precise in there:

```go
	flags, data := parseMountOptions(options)
	if len(data) >= pagesize {
		return errors.Errorf("can't mount: mount options string too long")
	}￼
￼```
or smth like that

Reply:
make senses. will update it.

Reply:
@kolyshkin I add another check after compact and show the error if the mount option is too long.
if we got empty `chdir` or the data is too long, return the `mount options is too long`. It is readable to user instead of `failed to mount: no such file or directory`.